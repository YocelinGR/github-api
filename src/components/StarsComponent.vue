<template>
  <div>
    <div
      v-for="repo in user.starredRepositories && user.starredRepositories.edges"
      :key="repo.node.id"
      class="star row"
    >
      <div class="col-md-8">
        <span class="userRepo">{{ repo.node.owner.login }} / </span>
        <a :href="repo.node.url" class="userRepo">{{ repo.node.name }}</a>
        <p class="description">{{ repo.node.description }}</p>
        <span
          v-for="topic in repo.node.repositoryTopics.nodes"
          :key="topic.id"
          class="topic"
        >
          {{ topic.topic.name }}
        </span>
        <span class="topic">{{ repo.starredAt }}</span>
      </div>
      <div class="col-md-3">
        <button class="btn btn-light" @click="removeStar(repo.node.id)"
          ><i class="far fa-star"></i> Unfollow</button
        >
      </div>
    </div>
  </div>
</template>
<script>
import gql from 'graphql-tag'

const starredByUser = gql`
  query user {
    user(login: "YocelinGR") {
      id
      starredRepositories(first: 10) {
        totalCount
        edges {
          starredAt
          node {
            id
            name
            description
            owner {
              login
              id
            }
            createdAt
            url
            repositoryTopics(first: 3) {
              nodes {
                id
                resourcePath
                url
                topic {
                  id
                  name
                }
              }
            }
          }
        }
      }
    }
  }
`
const removingStar = gql`
  mutation removeStar($input: RemoveStarInput!) {
    removeStar(input: $input) {
      clientMutationId
      starrable {
        id
      }
    }
  }
`
export default {
  data: () => ({
    user: {},
  }),
  methods: {
    removeStar(id) {
      const varStar = {
        starrableId: id,
        clientMutationId: 'MDQ6VXNlcjM5ODMzMDQ1',
      }
      this.$apollo.mutate({
        mutation: removingStar,
        variables: {
          input: varStar,
        },
      })
      // location.reload()
    },
  },
  apollo: {
    user: {
      query: starredByUser,
    },
  },
}
</script>
<style scoped>
.star {
  align-items: center;
  justify-content: center;
  padding-bottom: 10px;
  margin: 15px;
  border-bottom: rgb(187, 185, 185) solid 1px;
}
.btn {
  background-color: #f0f1f1;
  border-color: #a5a6a7;
}
.btn:hover {
  background-color: #ccd2db;
}
.userRepo {
  padding: 4px;
  font-size: 20px;
  color: rgb(10, 112, 196);
}
.description {
  padding: 2px;
  font-size: 18px;
  color: rgb(136, 137, 138);
}
.topic {
  padding: 6px;
  font-size: 14px;
  color: rgb(70, 71, 71);
}
</style>
