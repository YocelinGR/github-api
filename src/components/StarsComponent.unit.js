import StarsComponent from './StarsComponent.vue'
import { createLocalVue } from '@vue/test-utils'

describe('@components/StarsComponent', () => {
  const localVue = createLocalVue()
  it('StarsComponents is a vue instance', () => {
    const wrapper = shallowMount(StarsComponent, { localVue })
    expect(wrapper.isVueInstance()).toBeTruthy()
  })

  it('called Apollo mutations', () => {
    const mutate = jest.fn()
    const wrapper = mount(StarsComponent, {
      localVue,
      mocks: {
        $apollo: {
          mutate,
        },
      },
    })
    wrapper.vm.removeStar()
    expect(mutate).toBeCalled()
  })

  it('Displays user starred repositories data', () => {
    const wrapper = shallowMount(StarsComponent, { localVue })
    wrapper.setData({
      user: {
        starredRepositories: {
          totalCount: 2,
          edges: [
            {
              starredAt: '2019-02-15T15:45:51Z',
              node: {
                id: 'MDEwOlJlcG9zaXRvcnk5MjExMjMwOA==',
                name: 'javascript',
                description: null,
                owner: {
                  login: 'Jonhks',
                  id: 'MDQ6VXNlcjI1MzI4NjM1',
                },
                createdAt: '2017-05-23T00:43:46Z',
                url: 'https://github.com/Jonhks/javascript',
                repositoryTopics: {
                  nodes: [],
                },
              },
            },
            {
              starredAt: '2019-02-21T19:32:22Z',
              node: {
                id: 'MDEwOlJlcG9zaXRvcnkxNTIyODY3NzE=',
                name: 'YocelinGR.github.io',
                description:
                  'En el presente repositorio se presenta el portafolio personal de quien suscribe, en él se vierte la información personal y  de contacto, el background y los principales proyectos en los que se a trabajado. ',
                owner: {
                  login: 'YocelinGR',
                  id: 'MDQ6VXNlcjM5ODMzMDQ1',
                },
                createdAt: '2018-10-09T16:44:16Z',
                url: 'https://github.com/YocelinGR/YocelinGR.github.io',
                repositoryTopics: {
                  nodes: [
                    {
                      id: 'MDE1OlJlcG9zaXRvcnlUb3BpYzk0OTA1OTg=',
                      resourcePath: '/topics/javascript',
                      url: 'https://github.com/topics/javascript',
                      topic: {
                        id: 'MDU6VG9waWNqYXZhc2NyaXB0',
                        name: 'javascript',
                      },
                    },
                    {
                      id: 'MDE1OlJlcG9zaXRvcnlUb3BpYzk0OTA2MDA=',
                      resourcePath: '/topics/jest-tests',
                      url: 'https://github.com/topics/jest-tests',
                      topic: {
                        id: 'MDU6VG9waWNqZXN0LXRlc3Rz',
                        name: 'jest-tests',
                      },
                    },
                    {
                      id: 'MDE1OlJlcG9zaXRvcnlUb3BpYzk0OTA2MDI=',
                      resourcePath: '/topics/bootstrap4',
                      url: 'https://github.com/topics/bootstrap4',
                      topic: {
                        id: 'MDU6VG9waWNib290c3RyYXA0',
                        name: 'bootstrap4',
                      },
                    },
                  ],
                },
              },
            },
          ],
        },
      },
    })
    expect(wrapper.element).toMatchSnapshot()
  })
})
