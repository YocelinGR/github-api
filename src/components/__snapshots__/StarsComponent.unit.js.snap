// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`@components/StarsComponent Displays user starred repos data 1`] = `
<div>
  <div
    class="star row"
  >
    <div
      class="col-md-8"
    >
      <span
        class="userRepo"
      >
        Jonhks / 
      </span>
       
      <a
        class="userRepo"
        href="https://github.com/Jonhks/javascript"
      >
        javascript
      </a>
       
      <p
        class="description"
      >
        
      </p>
        
      <span
        class="topic"
      >
        2019-02-15T15:45:51Z
      </span>
    </div>
     
    <div
      class="col-md-3"
    >
      <button
        class="btn btn-light"
      >
        <i
          class="far fa-star"
        />
         Unfollow
      </button>
    </div>
  </div>
  <div
    class="star row"
  >
    <div
      class="col-md-8"
    >
      <span
        class="userRepo"
      >
        YocelinGR / 
      </span>
       
      <a
        class="userRepo"
        href="https://github.com/YocelinGR/YocelinGR.github.io"
      >
        YocelinGR.github.io
      </a>
       
      <p
        class="description"
      >
        En el presente repositorio se presenta el portafolio personal de quien suscribe, en él se vierte la información personal y  de contacto, el background y los principales proyectos en los que se a trabajado. 
      </p>
       
      <span
        class="topic"
      >
        
          javascript
        
      </span>
      <span
        class="topic"
      >
        
          jest-tests
        
      </span>
      <span
        class="topic"
      >
        
          bootstrap4
        
      </span>
       
      <span
        class="topic"
      >
        2019-02-21T19:32:22Z
      </span>
    </div>
     
    <div
      class="col-md-3"
    >
      <button
        class="btn btn-light"
      >
        <i
          class="far fa-star"
        />
         Unfollow
      </button>
    </div>
  </div>
</div>
`;

exports[`@components/StarsComponent Displays user starred repositories data 1`] = `
<div>
  <div
    class="star row"
  >
    <div
      class="col-md-8"
    >
      <span
        class="userRepo"
      >
        Jonhks / 
      </span>
       
      <a
        class="userRepo"
        href="https://github.com/Jonhks/javascript"
      >
        javascript
      </a>
       
      <p
        class="description"
      >
        
      </p>
        
      <span
        class="topic"
      >
        2019-02-15T15:45:51Z
      </span>
    </div>
     
    <div
      class="col-md-3"
    >
      <button
        class="btn btn-light"
      >
        <i
          class="far fa-star"
        />
         Unfollow
      </button>
    </div>
  </div>
  <div
    class="star row"
  >
    <div
      class="col-md-8"
    >
      <span
        class="userRepo"
      >
        YocelinGR / 
      </span>
       
      <a
        class="userRepo"
        href="https://github.com/YocelinGR/YocelinGR.github.io"
      >
        YocelinGR.github.io
      </a>
       
      <p
        class="description"
      >
        En el presente repositorio se presenta el portafolio personal de quien suscribe, en él se vierte la información personal y  de contacto, el background y los principales proyectos en los que se a trabajado. 
      </p>
       
      <span
        class="topic"
      >
        
        javascript
      
      </span>
      <span
        class="topic"
      >
        
        jest-tests
      
      </span>
      <span
        class="topic"
      >
        
        bootstrap4
      
      </span>
       
      <span
        class="topic"
      >
        2019-02-21T19:32:22Z
      </span>
    </div>
     
    <div
      class="col-md-3"
    >
      <button
        class="btn btn-light"
      >
        <i
          class="far fa-star"
        />
         Unfollow
      </button>
    </div>
  </div>
</div>
`;
